common/libs/wallet/non-deterministic/helpers.ts

```diff
@@ -33,22 +33,33 @@ export const signWrapper = (walletToWrap: IFullWallet): WrappedWallet =>
   });
 
 function determineKeystoreType(file: string): string {
-  const parsed = JSON.parse(file);
-  if (parsed.encseed) {
-    return KeystoreTypes.presale;
-  } else if (parsed.Crypto || parsed.crypto) {
-    return KeystoreTypes.utc;
-  } else if (parsed.hash && parsed.locked === true) {
-    return KeystoreTypes.v1Encrypted;
-  } else if (parsed.hash && parsed.locked === false) {
-    return KeystoreTypes.v1Unencrypted;
-  } else if (parsed.publisher === 'MyEtherWallet') {
-    return KeystoreTypes.v2Unencrypted;
-  } else {
-    throw new Error('Invalid keystore');
+  try {
+    const parsed = JSON.parse(file);
+    if (parsed.encseed) {
+      return KeystoreTypes.presale;
+    } else if (parsed.Crypto || parsed.crypto) {
+      return KeystoreTypes.utc;
+    } else if (parsed.hash && parsed.locked === true) {
+      return KeystoreTypes.v1Encrypted;
+    } else if (parsed.hash && parsed.locked === false) {
+      return KeystoreTypes.v1Unencrypted;
+    } else if (parsed.publisher === 'MyEtherWallet') {
+      return KeystoreTypes.v2Unencrypted;
+    } else {
+      return '';
+    }
+  } catch {
+    return '';
   }
 }
 
+const isKeystoreValid = (file: string): boolean => {
+  if (determineKeystoreType(file) === '') {
```

Comment:
Ah, it's here. This seems quite unintuitive though.

Reply:
Yea it isn't very intuitive, the error handling still could use some polish. Errors thrown by parsing an invalid file as JSON in `determineKeystoreType` make it difficult to write something easier to use. I spent some time on it the issue but didn't make much progress. I don't think it's a blocking issue, and it is still functional, so I'm going to come back to it later after I've fixed more critical issues.